### PR TITLE
Fixes IAM role permissions for PreheatJob

### PR DESCRIPTION
**Pull request by heythisisnate**
This is a 🐞 bug fix.

- [x] I've added tests (if it's a bug, feature or enhancement)
- [ ] I've adjusted the documentation (if it's a feature or enhancement)
- [x] The test suite passes (run `bundle exec rspec` to verify this)

## Summary

Fixes the default IAM policy for Jets::PreheatJob

## Context

The IAM role for PreheatJob appears to have been broken in c36f87f15e42c91991a867c2b6b10d4054b0603e, resulting in errors on Lambda like this:

> Aws::Lambda::Errors::AccessDeniedException User: arn:aws:sts::652695076726:assumed-role/bot-dev-JetsPreheatJob-1OYM89-JetsPreheatJobIamRole-C7QgfgQgszTl/bot-dev-jets-preheat_job-warm is not authorized to perform: lambda:InvokeFunction on resource: arn:aws:lambda:us-east-1:652695076726:function:bot-dev-admin_controller-db because no identity-based policy allows the lambda:InvokeFunction action

Basically, the generated `app-name-preheat-job-policy` was missing permission to invoke Lambda functions.

## How to Test

View the generated policy document in the Lambda console for the execution role associated with the preheat_job-warm function, and verify that it includes lambda:InvokeFunction.

## Version Changes

patch version bump to 4.0.6

### CODE REVIEW

**Comment by jeremiahlukus** [comment]
Ran into this issue as well thank you for sending a PR @heythisisnate !  You are right this should be included.

**Comment by tongueroo** [comment]
Released in 4.0.6

**Comment by heythisisnate** [comment]
uh-oh. I just tried to deploy after bumping 4.0.6 and got this failure:
> 09:07:51AM UPDATE_FAILED AWS::CloudFormation::Stack JetsPreheatJob Template format error: Unresolved resource dependencies [JetsPreheatJobWarmLambdaFunction] in the Resources block of the template

I'm not sure how this broke, but I'm looking into it.

**Comment by tongueroo** [comment]
All good. Thanks for following up and letting me know. Fixed in #671. 

Note: The Jets::PreheatJob Lambda Function Logic id will use the new format in Jets v5.
